taskflows/taskflows/__init__.py

```diff
@@ -491,26 +459,22 @@ def _create_job(task, input_, cluster, image, run_parameters, container_descript
 
     mount_option = '%s %s:%s' % (bind_args[container], host_dir, guest_dir)
 
-    if container == 'docker':
-        # In the docker case we need to ensure the image has been pull on this
-        # node, as the images are not shared across the nodes.
```

Comment:
This looks like something we will need to take into account if we do not automatically pull images.

Reply:
@cryos For the work that tries to separate out "docker pull" from the tasks, the above is a potential issue that Chris and I discussed earlier.

I guess part of the reason "docker pull" was originally implemented in the tasks is because, if we are running a task with multiple nodes, each node needs to have its own copy of the image (this is true for singularity as well if the nodes do not share a filesystem).

However, we think we could still move this to some kind of management interface in the client, and if the user chooses to update all docker images, a task could be launched that runs "docker pull" (and the corresponding singularity commands) on every node. Does this sound reasonable?

Reply:
This sounds reasonable, and I think it would be a good path forward.